cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -327,4 +342,250 @@ public HttpRequest rewriteRequest(HttpRequest request) {
       "/namespaces/" + Constants.DEFAULT_NAMESPACE));
     return request;
   }
+
+
+  protected final void dataList(HttpRequest request, HttpResponder responder, Store store, DatasetFramework dsFramework,
+                        Data type, String name, String appId) {
+    try {
+      if ((name != null && name.isEmpty()) || (appId != null && appId.isEmpty())) {
+        responder.sendString(HttpResponseStatus.BAD_REQUEST, "Empty name provided");
+        return;
+      }
+
+      String accountId = getAuthenticatedAccountId(request);
```

Comment:
This hardcoding shouldn't be here, no? Have the namespaceId passed to the method and maybe use the hardcoding in the caller.